crates/edr_napi/src/provider.rs

```diff
@@ -0,0 +1,58 @@
+mod config;
+
+use std::sync::Arc;
+
+use napi::{tokio::runtime, Env, JsObject, Status};
+use napi_derive::napi;
+
+use self::config::ProviderConfig;
+
+/// A JSON-RPC provider for Ethereum.
+#[napi]
+pub struct Provider {
+    provider: Arc<edr_provider::Provider>,
+}
+
+#[napi]
+impl Provider {
+    #[doc = "Constructs a new provider with the provided configuration."]
+    #[napi(ts_return_type = "Promise<Provider>")]
+    pub fn with_config(env: Env, config: ProviderConfig) -> napi::Result<JsObject> {
+        let config = edr_provider::ProviderConfig::try_from(config)?;
+        let runtime = runtime::Handle::current();
+
+        let (deferred, promise) = env.create_deferred()?;
+        runtime.clone().spawn(async move {
```

Comment:
Just for my understanding: what's the reason for cloning the runtime here?

Reply:
The `runtime` is moved into the the `async` callback